exercise3.4/RomanNumber.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package romannumberconversion;
+
+/**
+ *
+ * @author roberta
+ */
+public enum RomanNumber {
+
+    I(1),
+    V(5),
+    X(10),
+    L(50),
+    C(100),
+    D(500),
+    M(1000);
+
+    private final int value;
+
+    RomanNumber(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return this.value;
+    }
+
+    /**
+     * Devuelve el valor numérico de la letra romana
+     *
+     * @param l
+     * @return
+     */
+    public static int parse(char l) {
+        switch (l) {
+            case 'I':
+                return I.value;
+            case 'V':
+                return V.value;
```

Comment:
en ves de tener un switch se podria tener un bucle while que vaya comprando e iterando y corte cuando encuentra el elemento. Existe un metodo que se llama enumClass.values()

public static RomanNumber getRomanValueFromChar(Char i) {
for (RomanNumber romanNumber : RomanNumber.values()) {
if (romanNumber.getValue().equals(i)) {
return romanNumber;
}
}
// throw an IllegalArgumentException or return null
throw new IllegalArgumentException("the given letter doesn't match any Roman Number.");
}